include/singa/core/platform.h

```diff
@@ -0,0 +1,105 @@
+/**
```

Comment:
can you move the platform class back to device.h?
because other code uses include "device.h" to get the Platform class.